keyboards/hhkb_yang/MEMO.md

```diff
@@ -0,0 +1,94 @@
+## Hardware Information
+
+The YANG HHKB BLE controller design is similiar to hasu's
+controller. Most pins are compatiable.
+
+**MCU**: ATmega32U4  
+**Bluetooth**:  MDBT40 (nRF51822-based), with Adafruit Bluefruit LE UART Friend firmware.  
+**Power**: 3.3V  
+**CPU Frequency**: 8MHz  
+**Bootloader**: Lufa MassStorage  
+
+## Pin usage
+
+| Description                          | HASU pin usage         | YANG mod changed           |
+|:------------------------------------ | ---------------------- | -------------------------- |
+| ~KEY: Lo(0) when key is pressed      | PD7 input(with pullup) |                            |
+| Hysteresis: Hi(1) if key was pressed | PB7 output             |                            |
+| Row selector bit0                    | PB0 output             |                            |
+| Row selector bit1                    | PB1 output             |                            |
+| Row selector bit2                    | PB2 output             |                            |
+| Col selector bit0                    | PB3 output             |                            |
+| Col selector bit1                    | PB4 output             |                            |
+| Col selector bit2                    | PB5 output             |                            |
+| Key unable                           | PB6 output             |                            |
+| Switch power                         | PD4 output             | PD6 output (PMOS FET)      |
+| Bluetooth UART Rx                    | PC4 input              | PD2                        |
+| Bluetooth UART Tx                    | PC5 output             | PD3                        |
+| Bluetooth power                      |                        | PD5 output (low: power on) |
+| LED 0                                |                        | PF4                        |
+| LED 2                                |                        | PF1                        |
+| LED 4                                |                        | PF0                        |
+| Unused for PRO2                      | PC6                    |                            |
+| Unused for PRO2                      | PC7                    |                            |
+| Inner USB power                      |                        | PF7                        |
+
+## How to flash LUFA MassStorage bootloader on Linux
+
+The FAT filesystem on Linux very often cannot flush the write cache,
+leading to broken firmware in the flash.
+
+We can use `dd` to write to the virtual block storage directly to
+bypass the vfs layer.
+
+```
+dd if=FLASH.bin of=/dev/sda seek=4
```

Comment:
```suggestion

```
This is a nice and easy way to break most users installation.
Given that for most systems `/dev/sda` is the main drive and ppl are prone to blindly running commands from the internet, this is a disaster in the making.

If simply calling `sync` after a `cp` is not an option, I'd rather have a short guide here on finding the appropriate device (with `lsblk` or something) than this command.